### ISSUE TITLE
bug: Ionic React v6 does not respect `swipeBackEnabled: false` config option

### ISSUE THREAD

**Issue by thomasklemm** [ionic-team/ionic-framework#24273]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

In Ionic React v5, the `swipeBackEnabled: false` config option works as expected (disabling the swipe back gesture, which is enabled by default). 

![image](https://user-images.githubusercontent.com/1100176/143420292-16a4f49b-fc56-417b-b927-9efa8e6e8435.png)

When upgrading the app to Ionic React v6.0.0-rc.3 however, this config setting doesn't have an effect anymore. The swipe back gesture is enabled regardless of the `swipeBackEnabled: false`.

### Expected Behavior

According to the Ionic React v6 docs the config setting `swipeBackEnabled: false` should still exist and work: [Ionic React v6 Config - Config Options](https://beta.ionicframework.com/docs/react/config#config-options)

### Steps to Reproduce

I've created a reproduction repo here: [Repo](https://github.com/thomasklemm/IonicReactSwipeBackBug). The steps are visible in the [commit history](https://github.com/thomasklemm/IonicReactSwipeBackBug/commits/main).

- 1) At first a fresh Ionic React v5 app with the tabs layout is created from the starter app
- 2) By default, swipe back gesture is enabled: 
- 3) Swipe back gesture in Ionic React v5 gets disabled with `swipeBackEnabled: false`
- 4) Following the upgrade steps for Ionic React v6 beta (upgrading to the latest `6.0.0-rc.3`). Running the app afterwards causes the swipe back gesture to be enabled again, the config settings doesn't have an effect anymore.

### Code Reproduction URL

https://github.com/thomasklemm/IonicReactSwipeBackBug

### Ionic Info

```
Ionic:

   Ionic CLI       : 6.18.1 (/Users/thomasklemm/.nvm/versions/node/v16.13.0/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/react 6.0.0-rc.3

Capacitor:

   Capacitor CLI      : 3.3.2
   @capacitor/android : not installed
   @capacitor/core    : 3.3.2
   @capacitor/ios     : 3.3.2

Utility:

   cordova-res : 0.15.3
   native-run  : 1.5.0

System:

   NodeJS : v16.13.0 (/Users/thomasklemm/.nvm/versions/node/v16.13.0/bin/node)
   npm    : 8.1.2
   OS     : macOS Big Sur
```

**Comment by sean-perkins** [comment]
@thomasklemm thanks for reporting this issue!

This has been resolved in #24254 and will be available in the next release of Ionic 6. 

This is unfortunately a breaking change and will require you to update your code from:

```ts
import { setupConfig } from '@ionic/react';

setupConfig({
  swipeBackEnabled: false,
});
```

to:

```ts
import { setupIonicReact } from '@ionic/react';

setupIonicReact({
  swipeBackEnabled: false,
});
```

I will leave this issue open in the interim, in case others come across it.

**Comment by thomasklemm** [comment]
@sean-perkins Tried it with the latest `@ionic/react@dev` release, works!
